lib/zhuyin.ex

```diff
@@ -0,0 +1,376 @@
+    pinyin_initial = Map.get(@initials, zhuyin.initial, "")
+
+    pinyin_final =
+      if zhuyin.initial == "" do
+        @standalone_finals[zhuyin.final]
+      else
+        @finals[zhuyin.final]
+      end
+
+    %Pinyin{initial: pinyin_initial, final: pinyin_final, tone: zhuyin.tone}
+  end
+
+  def to_pinyin(list) when is_list(list) do
+    list
+    |> Enum.map(fn
+      z = %Zhuyin{} -> Zhuyin.to_pinyin(z)
+      str when is_binary(str) -> str
+    end)
+  end
+
+  @doc """
+  Create a pinyin struct from a zhuyin struct or list.
+
+  ## Examples
+
+      iex> Zhuyin.from_pinyin(~p/nǐhǎo/)
+      ~z/ㄋㄧˇㄏㄠˇ/
+
+      iex> Zhuyin.from_pinyin(%Pinyin{initial: "n", final: "i", tone: 3})
+      %Zhuyin{initial: "ㄋ", final: "ㄧ", tone: 3}
+
+  """
+  @spec from_pinyin(Pinyin.t() | Pinyin.pinyin_list()) :: t()
+  def from_pinyin(pinyin = %Pinyin{}) do
+    initial_map = Map.new(@initials, fn {key, val} -> {val, key} end)
+    initial = Map.get(initial_map, pinyin.initial, "")
+
+    final =
+      if pinyin.initial == "" do
+        standalone_finals_map = Map.new(@standalone_finals, fn {key, val} -> {val, key} end)
+        standalone_finals_map[pinyin.final]
+      else
+        finals_map = Map.new(@finals, fn {key, val} -> {val, key} end)
+        finals_map[pinyin.final]
+      end
+
+    %__MODULE__{initial: initial, final: final, tone: pinyin.tone}
+  end
+
+  def from_pinyin(list) when is_list(list) do
+    list
+    |> Enum.map(fn
+      p = %Pinyin{} -> from_pinyin(p)
+      str when is_binary(str) -> str
+    end)
+  end
+
+  def create(initial, final, tone) do
+    %__MODULE__{initial: initial, final: final, tone: tone_index(tone)}
+  end
+
+  def create(final, tone) do
+    %__MODULE__{final: final, tone: tone_index(tone)}
+  end
+
+  @spec read(String.t(), :exclusive | :words | :mixed) ::
+          {:ok, zhuyin_list()} | {:error, String.t()}
+  def read(string, mode \\ :exclusive) when mode in [:exclusive, :words, :mixed] do
+    res =
+      case mode do
+        :exclusive -> Zhuyin.Parsers.zhuyin_only(string)
+        :words -> Zhuyin.Parsers.zhuyin_words(string)
+        :mixed -> Zhuyin.Parsers.zhuyin_mix(string)
+      end
+
+    case res do
+      {:ok, lst, "", %{}, _, _} -> {:ok, lst}
+      {:error, _, remainder, %{}, _, _} -> {:error, remainder}
+    end
+  end
+
+  @spec read!(String.t(), :exclusive | :words | :mixed) ::
+          zhuyin_list() | no_return()
+  def read!(string, mode \\ :exclusive)
+      when mode in [:exclusive, :words, :mixed] do
+    case read(string, mode) do
+      {:ok, res} -> res
+      {:error, remainder} -> raise ParseError, remainder
+    end
+  end
+
+  defp parser_result({:ok, res, "", %{}, _, _}), do: {:ok, res}
+  defp parser_result({:error, _, rem, %{}, _, _}), do: {:error, rem}
+
+  defp parser_result!({:ok, res, "", %{}, _, _}), do: res
+  defp parser_result!({:error, _, rem, %{}, _, _}), do: raise(ParseError, rem)
+
+  @doc """
+  Create a zhuyin struct (`t:t/0`) from a string.
```

Comment:
I would use the word "single" here, to make this explicit in the summary generated at the top of the docs.

Reply:
Yes, sorry for not being more clear. The first line is shown in the summary so I think the extra context is useful there.

Reply:
Oh I see. Makes sense.